src/write_influxdb_udp.c

```diff
@@ -0,0 +1,701 @@
+    BUFFER_ADD(",type_instance=%s", temp);
+  }
+
+  BUFFER_ADD(" ");
+  for (size_t i = 0; i < ds->ds_num; i++) {
+    if ((ds->ds[i].type != DS_TYPE_COUNTER) &&
+        (ds->ds[i].type != DS_TYPE_GAUGE) &&
+        (ds->ds[i].type != DS_TYPE_DERIVE) &&
+        (ds->ds[i].type != DS_TYPE_ABSOLUTE)) {
+      sfree(rates);
+      return -1;
+    }
+
+    if (ds->ds[i].type == DS_TYPE_GAUGE) {
+      if (isnan(vl->values[i].gauge))
+        continue;
+      if (have_values)
+        BUFFER_ADD(",");
+      BUFFER_ADD("%s=%lf", ds->ds[i].name, vl->values[i].gauge);
+      have_values = true;
+    } else if (wifxudp_config_store_rates) {
+      if (rates == NULL)
+        rates = uc_get_rate(ds, vl);
+      if (rates == NULL) {
+        WARNING("write_influxdb_udp plugin: "
+                "uc_get_rate failed.");
+        return -1;
+      }
+      if (isnan(rates[i]))
+        continue;
+      if (have_values)
+        BUFFER_ADD(",");
+      BUFFER_ADD("%s=%lf", ds->ds[i].name, rates[i]);
+      have_values = true;
+    } else if (ds->ds[i].type == DS_TYPE_COUNTER) {
+      if (have_values)
+        BUFFER_ADD(",");
+      BUFFER_ADD("%s=%" PRIu64 "i", ds->ds[i].name,
+                 (uint64_t)vl->values[i].counter);
+      have_values = true;
+    } else if (ds->ds[i].type == DS_TYPE_DERIVE) {
+      if (have_values)
+        BUFFER_ADD(",");
+      BUFFER_ADD("%s=%" PRIi64 "i", ds->ds[i].name, vl->values[i].derive);
+      have_values = true;
+    } else if (ds->ds[i].type == DS_TYPE_ABSOLUTE) {
+      if (have_values)
+        BUFFER_ADD(",");
+      BUFFER_ADD("%s=%" PRIu64 "i", ds->ds[i].name, vl->values[i].absolute);
+      have_values = true;
+    }
+
+  } /* for ds->ds_num */
+  sfree(rates);
+
+  if (!have_values) {
+    return 0;
+  }
+  BUFFER_ADD(" %" PRIu64 "\n", CDTIME_T_TO_MS(vl->time));
+
+#undef BUFFER_ESCAPE
+#undef BUFFER_ADD
+
+  return offset;
+} /* int write_influxdb_point */
+
+static int
+write_influxdb_udp_write(const data_set_t *ds, const value_list_t *vl,
+                         user_data_t __attribute__((unused)) * user_data) {
+  int status;
+  char *values;
+  int values_len;
+
+  /* listen_loop is set to non-zero in the shutdown callback, which is
+   * guaranteed to be called *after* all the write threads have been shut
+   * down. */
+  assert(listen_loop == 0);
+
+  if (wifxudp_config_packet_size < NET_DEFAULT_PACKET_SIZE)
+    values_len = wifxudp_config_packet_size;
+  else
+    values_len = NET_DEFAULT_PACKET_SIZE;
+
+  values = malloc(values_len);
+  if (values == NULL) {
+    ERROR("write_influxdb_udp plugin: malloc failed.");
+    return -1;
+  }
+
+  status = write_influxdb_point(values, values_len, ds, vl);
+  if (status == 0) {
+    /* no real values to send (nan) */
+    free(values);
+    return 0;
+  }
+  if (status < 0) {
+    ERROR("write_influxdb_udp plugin: write_influxdb_point failed.");
+    free(values);
+    return -1;
+  } else if (status > wifxudp_config_packet_size) {
```

Comment:
I think this condition is not possible to occur.
`write_influxdb_point()` returns `-1` if it has no space in `values + values_len` defined buffer.
And `values_len <= wifxudp_config_packet_size` always.

Reply:
Absolutely right.